internal/csi-addons/rbd/volumegroup.go

```diff
@@ -363,3 +363,41 @@ func (vs *VolumeGroupServer) ModifyVolumeGroupMembership(
 		VolumeGroup: csiVG,
 	}, nil
 }
+
+// ControllerGetVolumeGroup RPC call to get a volume group.
+//
+// From the spec:
+// ControllerGetVolumeGroupResponse should contain current information of a
+// volume group if it exists. If the volume group does not exist any more,
+// ControllerGetVolumeGroup should return gRPC error code NOT_FOUND.
+func (vs *VolumeGroupServer) ControllerGetVolumeGroup(
+	ctx context.Context,
+	req *volumegroup.ControllerGetVolumeGroupRequest,
+) (*volumegroup.ControllerGetVolumeGroupResponse, error) {
+	mgr := rbd.NewManager(vs.csiID, nil, req.GetSecrets())
+	defer mgr.Destroy(ctx)
+
+	// resolve the volume group
+	vg, err := mgr.GetVolumeGroupByID(ctx, req.GetVolumeGroupId())
+	if err != nil {
+		return nil, status.Errorf(
+			codes.NotFound,
```

Comment:
@nixpanic what about the case where we have invalid secret or missing secret etc? do we need to consider it as not found?

Reply:
Actually, the only error that is documented in the spec is `NOT_FOUND`. The error description will contain more details on why the volume group could not be found.

Reply:
Lets revisit this later. 👍🏻